packages/goverlay/goverlay.pacscript

```diff
@@ -1,26 +1,37 @@
+maintainer="Xdavius <xdavius@github.com>"
 name="goverlay"
 gives="goverlay"
-pkgver="1.0"
-url="https://github.com/benjamimgois/goverlay/releases/download/${pkgver}/goverlay_${pkgver//.*}.tar.xz"
-hash="47754023d12b9eb730c83fe9aeec7b8523d1dbaa9860c90928ddaa904b89aa1f"
-depends=("mangohud" "libqt5pas-dev")
-pkgdesc="Graphical UI to help manage Linux overlays"
-maintainer="Elsie19 <elsie19@pm.me>"
+replace=("goverlay-bin")
+pkgver="1.1.1"
+pkgdesc="A GUI to help manage Vulkan/OpenGL overlays"
+arch=("amd64")
+makedepends=('git' 'libglu1-mesa-dev' 'qt6-base-dev')
+case "${DISTRO}" in
+debian:testing | debian:sid)
+  pacdeps=('qt6pas' 'mangohud')
+  makedepends+=("lazarus" "fpc" "fpc-source")
+  ;;
+*)
+  pacdeps=('mangohud' 'fpc-laz-deb' 'fpc-src-deb' 'lazarus-project-deb' 'qt6pas')
+  ;;
+esac
+url="https://github.com/benjamimgois/goverlay/archive/refs/tags/${pkgver}.tar.gz"
+hash="06dcad9ab7db990f2df04be01d549953e274907d7f691fe1bc4bbe712329936c"
 repology=("project: goverlay")
 
+build() {
+  case "${DISTRO}" in
+  debian:testing | debian:sid)
+    make -j "${NCPU}" LAZBUILDOPTS=--lazarusdir=/usr/lib/lazarus/default
+    ;;
+  *)
+    make -j "${NCPU}" LAZBUILDOPTS=--lazarusdir=/usr/share/lazarus/3.2.0/
+    ;;
+  esac
+}
+
 package() {
-  sudo mkdir -p "${pkgdir}/usr/bin/" "${pkgdir}/opt/goverlay/" "${pkgdir}/usr/share/applications/" "${pkgdir}/usr/share/icons/hicolor/128x128/apps/"
-  echo -e "#!/bin/sh\nQT_QPA_PLATFORM=xcb mangohud --dlsym /opt/goverlay/goverlay" | sudo tee "${pkgdir}/usr/bin/goverlay" > /dev/null
-  sudo chmod +x "${pkgdir}/usr/bin/goverlay"
-  sudo install -Dm755 goverlay "${pkgdir}/opt/goverlay/goverlay"
-  echo '[Desktop Entry]
-Name=GOverlay
-Comment=Graphical UI to help manage Vulkan / OpenGL overlays
-Exec=goverlay
-Icon=goverlay
-Terminal=false
-Type=Application
-Categories=Game;
-Keywords=MangoHud;vkBasalt;ReplaySorcery;msi;afterburner;' | sudo tee "${pkgdir}/usr/share/applications/goverlay.desktop" > /dev/null
-  sudo wget -q https://raw.githubusercontent.com/benjamimgois/goverlay/master/data/icons/128x128/goverlay.png -P "${pkgdir}/usr/share/icons/hicolor/128x128/apps/"
+  sudo make prefix=/usr libexecdir=/lib DESTDIR="${pkgdir}" install
+  # Fix Issue to change Depends with usage of pacdeps
+  sudo sed -i "/^Depends:*/c\Depends: mangohud, libqt6pas-dev" "${pkgdir}"/DEBIAN/control
```

Comment:
@Xdavius we're still deliberating on this line. That's why this hasn't been merged yet.

Reply:
Can be changed to prepare and install in package. You are right.

I will change it.

Reply:
@Elsie19 here the build of my friend Antiz for arch : 
https://gitlab.archlinux.org/archlinux/packaging/packages/goverlay/-/blob/main/PKGBUILD?ref_type=heads

And for my workaroud yes... I understand the problem.

We need a new section names `makepacdeps` to use pacstall as make dependency and not runtime dep.

For now, I suppose there is no other solution by editing the Debian Control file in this particular case.

On my repository, I have made worst because of missing pre_instal function :/ 

When pacstall will be ready for a new syntaxe, we will update the package and fix the workaround.

Other thing, I don't about pacstall 5.0 and the changes if it will be compatible... Well, if it break, we will patch XD